Feature - search bar filter

yuval-itzhak-hls opened the pull request:
Adding search pokemon bar with the custom SearchBar component.

yuval-itzhak-hls committed:
Adding search bar component to the home page.

Aslanchik-m commented on src/pages/home-page/PokemonsList.tsx:
```diff
@@ -94,17 +97,25 @@ export const PokemonsList: React.FC<PokemonsListProps> = ({
     : pokemons;
 
+
+   if (searchTerm?.trim()) {
+        const term = searchTerm.trim().toLowerCase();
+        filteredPokemons = filteredPokemons.filter(
+            (p) => p.name.toLowerCase().includes(term));
+   }
```
When does this happen?

yuval-itzhak-hls replied:
In the SearchBar component i use useDebouncedCallback to debounce the onSearch callback with a 300 ms delay. Currently, HomePage is passing setSearchTerm to SearchBars onChange prop, so searchTerm updates immediately on every key update, causing PokemonsList to rerender and run that filter block on each key update.

My intention was to pass setSearchTerm only to the debounced onSearch (not to onChange), so PokemonsList will rerender and filter only after 300 ms of no typing.

 I’ll update the code.

yuval-itzhak-hls committed:
refactor(SearchBar):
1) pulling out the debounce logic from the SearchBar component to the HomePage.
2)fix - making the search(the filter) debounced. before that the search action was excuted at every key update.
